com.hazelcast.jet.kafka.KafkaDataLinkTest

_master_ (commit 2b08d23caa99a542700755b6063007f2fa67bc00)

Failed on oracle-11: https://jenkins.hazelcast.com/job/Hazelcast-master-Kafka-confluent-platform-5.4.z/52/testReport/junit/com.hazelcast.jet.kafka/KafkaDataLinkTest/

`com.hazelcast.jet.kafka.KafkaDataLinkTest.list_resources_should_return_empty_list_for_no_topics`
<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: 

Expecting empty but was: [Resource{type='topic', name='__confluent.support.metrics'}]
	at com.hazelcast.jet.kafka.KafkaDataLinkTest.list_resources_should_return_empty_list_for_no_topics(KafkaDataLinkTest.java:98)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)

```
</details>


<details><summary>Standard output:</summary>

```
19:18:36,007  INFO || - [ProducerConfig] main - ProducerConfig values: 
	acks = 1
	batch.size = 16384
	bootstrap.servers = [PLAINTEXT://172.17.0.1:51203]
	buffer.memory = 33554432
	client.dns.lookup = use_all_dns_ips
	client.id = producer-67
	compression.type = none
	connections.max.idle.ms = 540000
	delivery.timeout.ms = 120000
	enable.idempotence = false
	interceptor.classes = []
	internal.auto.downgrade.txn.commit = false
	key.serializer = class org.apache.kafka.common.serialization.IntegerSerializer
	linger.ms = 0
	max.block.ms = 60000
	max.in.flight.requests.per.connection = 5
	max.request.size = 1048576
	metadata.max.age.ms = 300000
	metadata.max.idle.ms = 300000
	metric.reporters = []
	metrics.num.samples = 2
	metrics.recording.level = INFO
	metrics.sample.window.ms = 30000
	partitioner.class = class org.apache.kafka.clients.producer.internals.DefaultPartitioner
	receive.buffer.bytes = 32768
	reconnect.backoff.max.ms = 1000
	reconnect.backoff.ms = 50
	request.timeout.ms = 30000
	retries = 2147483647
	retry.backoff.ms = 100
	sasl.client.callback.handler.class = null
	sasl.jaas.config = null
	sasl.kerberos.kinit.cmd = /usr/bin/kinit
	sasl.kerberos.min.time.before.relogin = 60000
	sasl.kerberos.service.name = null
	sasl.kerberos.ticket.renew.jitter = 0.05
	sasl.kerberos.ticket.renew.window.factor = 0.8
	sasl.login.callback.handler.class = null
	sasl.login.class = null
	sasl.login.refresh.buffer.seconds = 300
	sasl.login.refresh.min.period.seconds = 60
	sasl.login.refresh.window.factor = 0.8
	sasl.login.refresh.window.jitter = 0.05
	sasl.mechanism = GSSAPI
	security.protocol = PLAINTEXT
	security.providers = null
	send.buffer.bytes = 131072
	socket.connection.setup.timeout.max.ms = 30000
	socket.connection.setup.timeout.ms = 10000
	ssl.cipher.suites = null
	ssl.enabled.protocols = [TLSv1.2, TLSv1.3]
	ssl.endpoint.identification.algorithm = https
	ssl.engine.factory.class = null
	ssl.key.password = null
	ssl.keymanager.algorithm = SunX509
	ssl.keystore.certificate.chain = null
	ssl.keystore.key = null
	ssl.keystore.location = null
	ssl.keystore.password = null
	ssl.keystore.type = JKS
	ssl.protocol = TLSv1.3
	ssl.provider = null
	ssl.secure.random.implementation = null
	ssl.trustmanager.algorithm = PKIX
	ssl.truststore.certificates = null
	ssl.truststore.location = null
	ssl.truststore.password = null
	ssl.truststore.type = JKS
	transaction.timeout.ms = 60000
	transactional.id = null
	value.serializer = class org.apache.kafka.common.serialization.StringSerializer

19:18:36,010  WARN || - [ProducerConfig] main - The configuration 'key.deserializer' was supplied but isn't a known config.
19:18:36,010  WARN || - [ProducerConfig] main - The configuration 'value.deserializer' was supplied but isn't a known config.
19:18:36,010  WARN || - [ProducerConfig] main - The configuration 'auto.offset.reset' was supplied but isn't a known config.
19:18:36,010  INFO || - [AppInfoParser] main - Kafka version: 2.8.2
19:18:36,010  INFO || - [AppInfoParser] main - Kafka commitId: 3146c6ff4a24cc24
19:18:36,010  INFO || - [AppInfoParser] main - Kafka startTimeMs: 1678648716010
19:18:36,010  INFO || - [AdminClientConfig] main - AdminClientConfig values: 
	bootstrap.servers = [PLAINTEXT://172.17.0.1:51203]
	client.dns.lookup = use_all_dns_ips
	client.id = 
	connections.max.idle.ms = 300000
	default.api.timeout.ms = 60000
	metadata.max.age.ms = 300000
	metric.reporters = []
	metrics.num.samples = 2
	metrics.recording.level = INFO
	metrics.sample.window.ms = 30000
	receive.buffer.bytes = 65536
	reconnect.backoff.max.ms = 1000
	reconnect.backoff.ms = 50
	request.timeout.ms = 30000
	retries = 2147483647
	retry.backoff.ms = 100
	sasl.client.callback.handler.class = null
	sasl.jaas.config = null
	sasl.kerberos.kinit.cmd = /usr/bin/kinit
	sasl.kerberos.min.time.before.relogin = 60000
	sasl.kerberos.service.name = null
	sasl.kerberos.ticket.renew.jitter = 0.05
	sasl.kerberos.ticket.renew.window.factor = 0.8
	sasl.login.callback.handler.class = null
	sasl.login.class = null
	sasl.login.refresh.buffer.seconds = 300
	sasl.login.refresh.min.period.seconds = 60
	sasl.login.refresh.window.factor = 0.8
	sasl.login.refresh.window.jitter = 0.05
	sasl.mechanism = GSSAPI
	security.protocol = PLAINTEXT
	security.providers = null
	send.buffer.bytes = 131072
	socket.connection.setup.timeout.max.ms = 30000
	socket.connection.setup.timeout.ms = 10000
	ssl.cipher.suites = null
	ssl.enabled.protocols = [TLSv1.2, TLSv1.3]
	ssl.endpoint.identification.algorithm = https
	ssl.engine.factory.class = null
	ssl.key.password = null
	ssl.keymanager.algorithm = SunX509
	ssl.keystore.certificate.chain = null
	ssl.keystore.key = null
	ssl.keystore.location = null
	ssl.keystore.password = null
	ssl.keystore.type = JKS
	ssl.protocol = TLSv1.3
	ssl.provider = null
	ssl.secure.random.implementation = null
	ssl.trustmanager.algorithm = PKIX
	ssl.truststore.certificates = null
	ssl.truststore.location = null
	ssl.truststore.password = null
	ssl.truststore.type = JKS

19:18:36,011  WARN || - [AdminClientConfig] main - The configuration 'key.deserializer' was supplied but isn't a known config.
19:18:36,011  WARN || - [AdminClientConfig] main - The configuration 'value.deserializer' was supplied but isn't a known config.
19:18:36,011  WARN || - [AdminClientConfig] main - The configuration 'value.serializer' was supplied but isn't a known config.
19:18:36,012  WARN || - [AdminClientConfig] main - The configuration 'auto.offset.reset' was supplied but isn't a known config.
19:18:36,012  WARN || - [AdminClientConfig] main - The configuration 'key.serializer' was supplied but isn't a known config.
19:18:36,012  INFO || - [AppInfoParser] main - Kafka version: 2.8.2
19:18:36,012  INFO || - [AppInfoParser] main - Kafka commitId: 3146c6ff4a24cc24
19:18:36,012  INFO || - [AppInfoParser] main - Kafka startTimeMs: 1678648716012
19:18:36,014  INFO || - [Metadata] kafka-producer-network-thread | producer-67 - [Producer clientId=producer-67] Cluster ID: g6QnoGPYR-WT3sSM139CyQ
19:18:36,038  INFO || - [AppInfoParser] kafka-admin-client-thread | adminclient-7 - App info kafka.admin.client for adminclient-7 unregistered
19:18:36,039  INFO || - [Metrics] kafka-admin-client-thread | adminclient-7 - Metrics scheduler closed
19:18:36,039  INFO || - [Metrics] kafka-admin-client-thread | adminclient-7 - Closing reporter org.apache.kafka.common.metrics.JmxReporter
19:18:36,039  INFO || - [Metrics] kafka-admin-client-thread | adminclient-7 - Metrics reporters closed
19:18:36,040  INFO || - [KafkaProducer] main - [Producer clientId=producer-67] Closing the Kafka producer with timeoutMillis = 9223372036854775807 ms.
19:18:36,041  INFO || - [Metrics] main - Metrics scheduler closed
19:18:36,041  INFO || - [Metrics] main - Closing reporter org.apache.kafka.common.metrics.JmxReporter
19:18:36,041  INFO || - [Metrics] main - Metrics reporters closed
19:18:36,041  INFO || - [AppInfoParser] main - App info kafka.producer for producer-67 unregistered

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-Kafka-confluent-platform-5.4.z/52/



`com.hazelcast.jet.kafka.KafkaDataLinkTest.list_resources_should_return_topics`:
<details><summary>Stacktrace:</summary>

```
org.junit.ComparisonFailure: 

Expecting actual:
  [Resource{type='topic', name='__confluent.support.metrics'},
    Resource{type='topic', name='my-topic'}]
to contain exactly (and in same order):
  [Resource{type='topic', name='my-topic'}]
but some elements were not expected:
  [Resource{type='topic', name='__confluent.support.metrics'}]
 expected:<...type='topic', name='[]my-topic'}]> but was:<...type='topic', name='[__confluent.support.metrics'},
    Resource{type='topic', name=']my-topic'}]>
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at com.hazelcast.jet.kafka.KafkaDataLinkTest.list_resources_should_return_topics(KafkaDataLinkTest.java:110)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)

```
</details>



Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-Kafka-confluent-platform-5.4.z/52/

--- a/extensions/kafka/src/test/java/com/hazelcast/jet/kafka/KafkaDataLinkTest.java
+++ b/extensions/kafka/src/test/java/com/hazelcast/jet/kafka/KafkaDataLinkTest.java
@@ -32,8 +32,11 @@ import org.junit.BeforeClass;
 import org.junit.Test;
 
 import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
 import java.util.Properties;
 
+import static java.util.stream.Collectors.toList;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.assertj.core.api.Fail.fail;
@@ -41,7 +44,7 @@ import static org.assertj.core.api.Fail.fail;
 
 public class KafkaDataLinkTest {
 
-    private static KafkaTestSupport kafkaTestSupport = KafkaTestSupport.create();
+    private static final KafkaTestSupport kafkaTestSupport = KafkaTestSupport.create();
     private KafkaDataLink kafkaDataLink;
 
     @BeforeClass
@@ -50,12 +53,12 @@ public class KafkaDataLinkTest {
     }
 
     @AfterClass
-    public static void afterClass() throws Exception {
+    public static void afterClass()  {
         kafkaTestSupport.shutdownKafkaCluster();
     }
 
     @After
-    public void tearDown() throws Exception {
+    public void tearDown() {
         if (kafkaDataLink != null) {
             kafkaDataLink.destroy();
         }
@@ -83,7 +86,7 @@ public class KafkaDataLinkTest {
 
     @Test
     public void non_shared_data_link_should_return_new_producer() {
-        kafkaDataLink = createNonSharedKafkaDataLink(kafkaTestSupport);
+        kafkaDataLink = createNonSharedKafkaDataLink();
 
         try (Producer<Object, Object> p1 = kafkaDataLink.getProducer(null);
              Producer<Object, Object> p2 = kafkaDataLink.getProducer(null)) {
@@ -95,7 +98,10 @@ public class KafkaDataLinkTest {
     public void list_resources_should_return_empty_list_for_no_topics() {
         kafkaDataLink = createKafkaDataLink(kafkaTestSupport);
 
-        assertThat(kafkaDataLink.listResources()).isEmpty();
+        Collection<DataLinkResource> resources = kafkaDataLink.listResources();
+        List<DataLinkResource> withoutConfluent =
+                resources.stream().filter(r -> !r.name().contains("__confluent")).collect(toList());
+        assertThat(withoutConfluent).isEmpty();
     }
 
     @Test
@@ -106,7 +112,11 @@ public class KafkaDataLinkTest {
         kafkaDataLink = createKafkaDataLink(localTestSupport);
         try {
             localTestSupport.createTopic("my-topic", 2);
-            assertThat(kafkaDataLink.listResources())
+
+            Collection<DataLinkResource> resources = kafkaDataLink.listResources();
+            List<DataLinkResource> withoutConfluent =
+                    resources.stream().filter(r -> !r.name().contains("__confluent")).collect(toList());
+            assertThat(withoutConfluent)
                     .containsExactly(new DataLinkResource("topic", "my-topic"));
         } finally {
             kafkaDataLink.destroy();
@@ -170,7 +180,7 @@ public class KafkaDataLinkTest {
         return new KafkaDataLink(config);
     }
 
-    private KafkaDataLink createNonSharedKafkaDataLink(KafkaTestSupport kafkaTestSupport) {
+    private KafkaDataLink createNonSharedKafkaDataLink() {
         DataLinkConfig config = new DataLinkConfig("kafka-data-link")
                 .setClassName(KafkaDataLink.class.getName())
                 .setShared(false)

